go/store/nbs/archive_chunk_source.go

```diff
@@ -152,3 +154,27 @@ func (acs archiveChunkSource) getRecordRanges(_ context.Context, _ []getRecord)
 func (acs archiveChunkSource) getManyCompressed(ctx context.Context, eg *errgroup.Group, reqs []getRecord, found func(context.Context, CompressedChunk), stats *Stats) (bool, error) {
 	return false, errors.New("Archive chunk source does not support getManyCompressed")
 }
+
+func (acs archiveChunkSource) getAllChunkHashes(ctx context.Context, out chan<- hash.Hash, wg *sync.WaitGroup) int {
+	addrCount := uint32(len(acs.aRdr.prefixes))
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for i := uint32(0); i < addrCount; i++ {
+			var h hash.Hash
+			suffix := acs.aRdr.getSuffixByID(i)
+
+			// Reconstruct the hash from the prefix and suffix.
+			binary.BigEndian.PutUint64(h[:uint64Size], acs.aRdr.prefixes[i])
+			copy(h[hash.ByteLen-hash.SuffixLen:], suffix[:])
```

Comment:
More readable as `copy(h[uint64Size:], suffix[:])`, since we just wrote the prefix into `h[:uint64Size]`?